java/src/shared/facades/BuildingFacade.java

```diff
@@ -0,0 +1,176 @@
+     * @param buildLocation  The edge on the map where the road is to be built.
+     * @param isFree  Whether the current player gets to build this for free.
+     */
+    public void buildRoad(EdgeLocation buildLocation, boolean isFree){}
+
+    /**
+     * Builds a settlement for the current player at the specified location.
+     * @pre canBuildSettlement returns true for the current player and the given location.
+     * @post Settlement is built at given location.  Current Player has one fewer unused settlements.  Current player gains one victory point.
+     * @param buildLocation  The hex vertex on the map where the settlement is to be built.
+     * @param isFree  Whether the current player gets to build this for free.
+     */
+    public void buildSettlement(VertexLocation buildLocation, boolean isFree){}
+
+    /**
+     * Builds a city for the current player at the specified location.
+     * @pre canBuildCity returns true for the current player and the given location
+     * @post City is built at given location.  Current player has one fewer unused cities and one more unused settlements.  Current player gains one victory point.
+     * @param buildLocation The hex vertex on the map where the city is to be built.
+     */
+    public void buildCity(VertexLocation buildLocation){}
+
+    /**
+     * Check to see if the current player is able to build a road
+     * @pre Player is currently in a game.
+     * @post None.
+     * @param buildLocation  The edge where the player wants to build a road.
+     * @param isFree  Whether the current player gets to build this for free.
+     * @return True if the player can build a road according to the rules of Catan; false otherwise.
+     */
+    public boolean canBuildRoad(EdgeLocation buildLocation, boolean isFree){
+        return false;
+    }
+
+    /**
+     * Check to see if the current player is able to build a settlement.
+     * @pre Player is currently in a game.
+     * @post None.
+     * @param buildLocation The vertex where the player wants to build a settlement.
+     * @param isFree  Whether the current player gets to build this for free.
+     * @return True if the player can build a settlement according to the rules of Catan; false otherwise.
+     */
+    public boolean canBuildSettlement(VertexLocation buildLocation, boolean isFree){
+        return false;
+    }
+
+    /**
+     * Check to see if the current player is able to build a settlement.
+     * @pre Player is currently in a game.
+     * @post None.
+     * @param buildLocation The vertex where the player wants to build a city.
+     * @return True if the payer can build a city according to the rules of Catan; false otherwise.
+     */
+    public boolean canBuildCity(VertexLocation buildLocation){
+        return false;
+    }
+
+    /**
+     * Get the number of roads a player has built.
+     * @pre The player is currently in a game.
+     * @post None
```

Comment:
Consistency in punctuation would be nice (but isn't really required). Feel free to ignore this comment.